### DISCUSSION TITLE
NextAuth (MongoDB & Mongoose) - Populate from Users MongoDB Collection?

### DISCUSSION THREAD

**Discussion by peterhogler** [nextauthjs/next-auth#8458] [Help]
So I have this case where I'm creating a new listing.

```
const listingSchema = new Schema<ListingSchemaDocument>({
     userId: { type: SchemaTypes.ObjectId, ref: "users", required: true },
     title: { type: String, required: true },
     genre: { type: String, required: true },
     description: { type: String, required: true },
     imageUrl: { type: String, required: true },
});
```

and wants to populate a field:

```
export const getAllListings = async () => {
    try {
        connectWithMongoDB();
        const listings = await Listing.find().populate("userId");
        return listings;
    } catch (error) {
        if (error instanceof Error) {
            throw new Error(`Error while trying to get all listings: ${error.message}`);
        }
    }
};
```

I get a error from NextJS saying in the lines of "Error: Error while trying to get all listings: Schema hasn't been registered for model "users".
Use mongoose.model(name, schema)" this is the same if I try to change the ref to "User", "Users", "users" etc.

Which is true, How can populate the userId field with a user from the users collection on mongoDB?
Since It's automatically made from the mongoDB adapter?

**Comment by grepsoft** [comment]
I am having the exact same issue and am unable to get the userid populated. Anyone has a solution?
